### ISSUE TITLE
WrapProcess doesn't work for cloned client

### ISSUE THREAD

**Issue by smacker** [go-redis/redis#471]
I track redis calls for every request. (I need to know how much time every endpoint consumed and how: go-itself/db/redis/external-http-requests)

In v4 I did this:
```
func WrapRedisClient(ctx context.Context, c *redis.Client) *redis.Client {
	txn := GetTnxFromContext(ctx)
	if txn == nil {
		return c
	}

	cCopy := *c
	cCopyP := &cCopy

	cCopyP.WrapProcess(RedisWrapper(txn))

	return cCopyP
}
```

So it measured time for redis for every transaction separately.

It doesn't work for v5 anymore because you set `client.cmdable.process = client.Process` in init, so `cmdable.process` points on original client. Do you think you can fix it? Or my case is out of your scope and I have to downgrade/fork?

**Comment by smacker** [comment]
if it will support context - it will be even better for my purposes. Thanks a lot!

**Comment by smacker** [comment]
fixed in #474
